### ISSUE TITLE
Cannot archive project with AWSCore when Bitcode is ENABLED

### ISSUE THREAD

**Issue by irlabs** [aws/aws-sdk-ios#301]
There is a problem with the latest version (2.3.3) of the AWS SDK for iOS and Bitcode.

When I

1. start a vanilla Xcode iOS swift project (with *Enable Bitcode = YES* by default),
2. add the `AWSCore.framework`,
3. use the `AWSCredentialsProvider` protocol simply like this:
    ```
import AWSCore

let provider = AWSAnonymousCredentialsProvider()
print(provider)
```
4. and then try to **Archive**,

Xcode fails and gives me this error:

```
ld: bitcode bundle could not be generated because '.../BitcodeBugExample/lib/
AWSCore.framework/AWSCore(AWSCredentialsProvider.o)' was built without full bitcode.
All object files and libraries for bitcode must be generated from Xcode Archive or
Install build for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

----

I've put an this vanilla example project (**with the above bug**) on github: [irlabs/AWSSDKBitcodeProblem](https://github.com/irlabs/AWSSDKBitcodeProblem)

----

When I build the `AWSCore.framework` myself with `OTHER_CFLAGS="-fembed-bitcode -Qunused-arguments"` in `Scripts/SdkPackage.sh` it *does* work.

I am Running Xcode 7.2

**Comment by yosuke-matsuda** [comment]
The frameworks seem to be built using `-fembed-bitcode-marker` instead of `-fembed-bitcode`. Thank you for bringing this issue to our attention. We will fix the build process. Thanks.

**Comment by mattijsf** [comment]
Similar to what I experienced in https://github.com/aws/aws-sdk-ios/issues/232#issuecomment-164435812. Solved it by doing the build myself with the same flags. But apparently the release binary (2.3.3) is indeed broken.

**Comment by hashitha** [comment]
I am having the same issue, where did you put the OTHER_CFLAGS="-fembed-bitcode -Qunused-arguments" in SdkPackage.sh

**Comment by hashitha** [comment]
Got it working. thanks

xcodebuild ARCHS="armv7 armv7s arm64 i386 x86_64" \
    OTHER_CFLAGS="-fembed-bitcode -Qunused-arguments" \
	ONLY_ACTIVE_ARCH=NO \
	-configuration Debug \
    -project "${project_path}/${project_name}.xcodeproj" \
    -target "${project_name}" \
    -sdk iphonesimulator \
    SYMROOT=$(PWD)/builtFramework \
    clean build

exitOnFailureCode $?

xcodebuild ARCHS="armv7 armv7s arm64 i386 x86_64" \
    OTHER_CFLAGS="-fembed-bitcode -Qunused-arguments" \
	ONLY_ACTIVE_ARCH=NO \
	-configuration Release \
    -project "${project_path}/${project_name}.xcodeproj" \
    -target "${project_name}" \
    -sdk iphoneos \
    SYMROOT=$(PWD)/builtFramework \
    clean build

**Comment by yosuke-matsuda** [comment]
We've released the AWS Mobile SDK for iOS 2.3.4 to address this issue? Could you see if the latest release fixes the issue?

**Comment by irlabs** [comment]
Yes, version 2.3.4. fixes the issue when Bitcode is enabled. Thanks!
